Update CHANGELOG.md (#1184)
nodejs/node-addon-api: CHANGELOG.md

#### Documentation

- Added documentation for `Napi::Object::PropertyLValue`.
- Changed all N-API references to Node-API.
- Some minor corrections all over the documentation.

#### TEST

> - Added tests relating to fetching property from Global Object.

Reviewer: I think this is about a `fetch` property and not about fetching properties, but I'm not sure. @nodejs/node-api 

```suggestion
- Added tests relating to `fetch` property from Global Object.
```

### TOOL

> - Setup Github actions for tests.

Reviewer: ```suggestion
- Setup GitHub Actions for tests.
```

- Added stale action.

> - Removed `Sudo` tag from Travis CI.

Reviewer: ```suggestion
- Removed `sudo` tag from Travis CI.
```

#### API

- Introduced `include_dir` for use with **gyp** in a scalar context.
- Added `Napi::Addon` to help handle the loading of a native add-on into
multiple threads and or multiple times in the same thread.
- Concentrate callbacks provided to core N-API.

> - Make sure wrap callback is used.

Reviewer: I think this refers to a `wrapcallback` property or somesuch but I'm not sure. @nodejs/node-api 

```suggestion
- Make sure `wrapcallback` is used.
```

#### TEST

- Added tests to check the build process.
- Refactored test for threadsafe function using async/await.

> - Converted tests that go into async functions that await 10 ticks after
> each go.

Reviewer: ```suggestion
- Converted tests that `gc` into async functions that await 10 ticks after
each `gc`.
```

#### TEST

> - An added test case for bool operator.

Reviewer: ```suggestion
- Added test case for bool operator.
```

#### API

- Added `SuppressDestruct` method to `Napi::AsyncWorker`.

> - Added new build targets for debugging.

Reviewer: Maybe this?

```suggestion
- Added new build targets for debug builds.
```

- Exposed macros that throw errors.

> - Fixed memory leaks caused by callback data when an api error occurs.

Reviewer: ```suggestion
- Fixed memory leaks caused by callback data when a Node-API error occurs.
```

#### Documentation

> - Completed the documentation to cover all the API surfaces.

Reviewer: ```suggestion
- Completed the documentation to cover all of the API surface.
```

  - Fixed initialization of std::string to nullptr

#### Tests

> - Fix test failures on Linux one and AIX

Reviewer: ```suggestion
- Fix test failures on LinuxOne and AIX
```

* [4058a29989] - n-api: fix memory leak in napi_async_destroy() (Jinho Bang) https://github.com/nodejs/node-addon-api/pull/213

> 

Reviewer: ```suggestion
````
